### PR TITLE
LOCAL DRAFT of WG Serving proposal

### PR DESCRIPTION
**Pull request by SergeyKanzhelev**
LOCAL DRAFT ONLY

### CODE REVIEW

File: sigs.yaml [hunk 1]
```diff
@@ -3467,6 +3467,47 @@ workinggroups:
     liaison:
       github: pohly
       name: Patrick Ohly
+- dir: wg-serving
+  name: Serving
+  mission_statement: >
+    Discuss and enhance the support of inference serving for accelerated workloads.
```
Comment (smarterclayton) [inline sigs.yaml]: nit: stray period at end of line

File: sigs.yaml [hunk 2]
```diff
@@ -3467,6 +3467,47 @@ workinggroups:
...
+    in Kubernetes. Making Kubernetes the natural choice when choosing the platform
```
Comment (smarterclayton) [inline sigs.yaml]: Suggestion for second sentence:

> Make Kubernetes the natural choice for hosting production inference reliably, and improve all serving workloads along the way.

File: wg-serving/charter.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+- Partner with existing ecosystem projects like kServe, Seldon, Kaito, and
+  others to identify, extract, or implement common shared problems (like Kueue
+  abstracted deferred scheduling for multiple batch frameworks).
+- Explore new projects that improve orchestration, scaling, and load balancing
+  of inference workloads and compose well with other workloads on Kubernetes
+
+### Out of scope
+
+- Batch inference, which should be covered by WG Batch.
+- Addition of new API kinds that serve a specialized type of workload. The focus
```
Comment (smarterclayton) [inline wg-serving/charter.md]: I don't know that this is out of scope.  Either way, it's vague.

File: wg-serving/charter.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+- Directly improve key kubernetes workload controllers when used with
+  accelerators and the most common inference serving frameworks and model
+  servers.
+- Partner with existing ecosystem projects like kServe, Seldon, Kaito, and
+  others to identify, extract, or implement common shared problems (like Kueue
+  abstracted deferred scheduling for multiple batch frameworks).
+- Explore new projects that improve orchestration, scaling, and load balancing
+  of inference workloads and compose well with other workloads on Kubernetes
+
+### Out of scope
```
Comment (smarterclayton) [inline wg-serving/charter.md:42]: We need to spend more time calling out out of scope items

File: wg-serving/charter.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+the Roles and Organization Management outlined in [wg-governance].
+
+[Kubernetes Charter README]: /committee-steering/governance/README.md
+
+## Scope
+
+Discuss and enhance the support for AI/ML inference workloads in Kubernetes.
+
+
+### In scope
```
Comment (smarterclayton) [inline wg-serving/charter.md:24]: Contrasting this to https://github.com/SergeyKanzhelev/community-1/blob/135df82737eba050d99268db7ed60b63baa12c9a/wg-batch/charter.md, i think we could tighten the in scope section up.  Maybe 1, 2, 4 could be one bullet, then we have another bullet with a suggestion of the areas that are in scope from the use cases doc?

File: sigs.yaml (hunk 1 above)
Comment (ahg-g) [inline sigs.yaml]: Do we need to limit the focus to accelerated workloads only?

File: wg-serving/charter.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+- Provide concrete input to other SIGs and WG around needs of inference
```
Comment (ahg-g) [inline wg-serving/charter.md]:
```suggestion
- Provide concrete input to other SIGs and WGs around needs of inference
```

File: wg-serving/charter.md (hunk 5 above)
Comment (ahg-g) [inline wg-serving/charter.md:24]: mixing of batch/training and inference could be in scope, or do we want to leave that to wg-batch?

File: wg-serving/charter.md (hunk 3 above)
Comment (ahg-g) [reply wg-serving/charter.md]: LWS is kinda that, so I don't think this should be out of scope

File: wg-serving/charter.md (hunk 5 above)
Comment (smarterclayton) [reply wg-serving/charter.md:24]: I think we should describe in scope as "being able to run serving workloads safely while giving up available slack capacity to batch frameworks"?

File: wg-serving/charter.md (hunk 3 above)
Comment (smarterclayton) [reply wg-serving/charter.md]: We should probably describe workflow for serving worlkoads as out of scope (i.e. we're not offering a MLOps platform, we're making Kubernetes better for the output of an MLOps pipeline)

File: sigs.yaml (hunk 1 above)
Comment (SergeyKanzhelev) [reply sigs.yaml]: second sentence now covers other scenarios too. I think concentrating on accelerators would make sense. Are you thinking of CPU serving as a fallback option as a scenario? Or pure CPU serving that needs to be improved?

File: wg-serving/charter.md (hunk 5 above)
Comment (SergeyKanzhelev) [reply wg-serving/charter.md:24]: added

File: wg-serving/charter.md (hunk 3 above)
Comment (SergeyKanzhelev) [reply wg-serving/charter.md]: added the workflows item. As for LWS - it is still generalized, no? I know it is written too vaguely and mostly copied from WG Batch charter. Changed to "Addition of new API kinds that serve a specific models. The focus should be on
  general APIs that frameworks can build on top of."

File: wg-serving/charter.md (hunk 4 above)
Comment (SergeyKanzhelev) [reply wg-serving/charter.md:42]: Some ideas, maybe too obvious to call out explicitly:

1. Maintain integrations with the specific frameworks
2. Security and privacy questions of the inference workloads beyond required extensibility points and monitoring capabilities
3. Questions of model testing and correctness beyond ability to deploy and execute test models.

Comment (SergeyKanzhelev) [reply wg-serving/charter.md:42]: workflows was a good call. Ease of integrating with plugins may be in scope?
